// user.proto
syntax = "proto3";

package users;

service AuthService {
    rpc Authenticate (AuthRequest) returns (AuthResponse);
    rpc ValidateToken (TokenRequest) returns (TokenResponse);
    rpc DestroyToken (TokenRequest) returns (DestroyTokenResponse);
}

message AuthRequest {
    string username = 1;
    string password = 2;
}

message AuthResponse {
    string token = 1;
}

message TokenRequest {
    string token = 1;
}

message TokenResponse {
    bool valid = 1;
    string message = 2;
    string sub = 3;
    string role = 4;
}

message DestroyTokenResponse {
    bool success = 1;
    string message = 2;
}

